### PR TITLE
[MDS] add dashboards sanity tests with multiple data source enabled

### PR DESCRIPTION
**Pull request by zhongnansu**
### Description
- create sanity tests with multiple data source support. Test cases are below
  - add sample data to datasource
  - check page can be loaded with expected content for dashboards, visualization, discover, stack management, index pattern and dev tools
- Add a new CI workflow to run MDS specific tests in core
- Add some util functions for mds: 
  - add/remove sample data against a data source, 
  - select data source by title from data source selector

### Issues Resolved

[List any issues this PR will resolve]

### Check List

- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/mds_dashboard_sanity_test_spec.js [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+    before(() => {});
+
+    after(() => {});
+
+    describe('checking home page', () => {
+      before(() => {
+        // Go to the home page
+        miscUtils.visitPage('app/home#');
+        cy.window().then((win) =>
+          win.localStorage.setItem('home:welcome:show', false)
```
Comment (zhyuanqi) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/mds_dashboard_sanity_test_spec.js:30]: Hi. Could you help me to understand why we add this home:welcome:show in localStorage?

**Review by zhyuanqi — APPROVED** [review]

File: .github/workflows/cypress-workflow-bundle-snapshot-based.yml [hunk 2]
```diff
@@ -26,3 +26,11 @@ jobs:
       test-command: env CYPRESS_NO_COMMAND_LOG=1 CYPRESS_ML_COMMONS_DASHBOARDS_ENABLED=true CYPRESS_VISBUILDER_ENABLED=true CYPRESS_DATASOURCE_MANAGEMENT_ENABLED=true yarn cypress:run-without-security --browser chromium --spec 'cypress/integration/core-opensearch-dashboards/opensearch-dashboards/**/*.js'
       osd-serve-args: --data_source.enabled=true --data_source.ssl.verificationMode=none --vis_builder.enabled=true --ml_commons_dashboards.enabled=true
       security-enabled: false
+
+  tests-with-multiple-data-source:
+    uses: ./.github/workflows/release-e2e-workflow-template.yml
+    with:
+      test-name: Core Dashboards using Bundle Snapshot
+      test-command: env CYPRESS_DISABLE_LOCAL_CLUSTER=true CYPRESS_DATASOURCE_MANAGEMENT_ENABLED=true yarn cypress:run-without-security --browser chromium --spec 'cypress/integration/core-opensearch-dashboards/opensearch-dashboards/**/mds*.js'
```
Comment (SuZhou-Joe) [inline .github/workflows/cypress-workflow-bundle-snapshot-based.yml:34]: Any reason we run the test `without` security?

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/mds_dashboard_sanity_test_spec.js [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+const path = baseURL.pathname.replace(/\/$/, '');
+
+const disableLocalCluster = !!Cypress.env('DISABLE_LOCAL_CLUSTER');
+
+if (Cypress.env('DATASOURCE_MANAGEMENT_ENABLED')) {
+  describe('dashboard data source sample data validation', () => {
+    before(() => {});
+
+    after(() => {});
+
```
Comment (SuZhou-Joe) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/mds_dashboard_sanity_test_spec.js:24]:
```suggestion
    before(() => {});

    after(() => {});
```
Useless code I suppose?

**Review by BionIT — APPROVED** [review]

**Comment by BionIT** [comment]
@zhongnansu do we know why the e2e workflow failed?

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/mds_dashboard_sanity_test_spec.js (hunk 3 above)
Comment (zhongnansu) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/mds_dashboard_sanity_test_spec.js]: removed

File: .github/workflows/cypress-workflow-bundle-snapshot-based.yml (hunk 2 above)
Comment (zhongnansu) [reply .github/workflows/cypress-workflow-bundle-snapshot-based.yml]: updated to run with security

**Comment by zhongnansu** [comment]
> @zhongnansu do we know why the e2e workflow failed?
@BionIT 
e2e workflow failed due to vis-builder tests, not related to change in this PR.
```
 ✖  core-opensearch-dashboards/opensear      01:33        5        4        1        -        - │
  │    ch-dashboards/apps/vis_builder/basi                                                         │
  │    c.spec.js                                                                                   │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  core-opensearch-dashboards/opensear      04:19        4        2        2        -        - │
  │    ch-dashboards/apps/vis_builder/dash                                                         │
  │    board.spec.js                                                                               │
```

I added a specific workflow to run tests for MDS with hide local cluster, and it passed.
https://github.com/opensearch-project/opensearch-dashboards-functional-test/actions/runs/9135444299/job/25122771055?pr=1330

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/mds_dashboard_sanity_test_spec.js (hunk 1 above)
Comment (zhongnansu) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/mds_dashboard_sanity_test_spec.js:31]: @zhyuanqi 
it's for hiding this welcome page
image

**Comment by zhongnansu** [comment]
@CCongWang @SuZhou-Joe Addressed all comments, could you review again?

**Review by SuZhou-Joe — APPROVED** [review]

**Review by CCongWang — APPROVED** [review]

**Pull request merged**